### DISCUSSION TITLE
MDN for Sustainable Computing Resources & Best Practices

### DISCUSSION THREAD

**Discussion by heaversm** [mdn/mdn-community#280] [Content]
Strategies for sustainable coding and infrastructure management are not well understood yet more essential than ever. 

In Google’s [2022 Cloud Sustainability Survey](https://cloud.google.com/blog/topics/sustainability/new-survey-reveals-executives-views-about-sustainability), 65% of IT executives want to improve sustainability efforts but don’t know how to do it and 36% say they lack tools to track progress toward sustainability goals, and just 17% say they are using those measurements to optimize based on results. Google predicts that by 2025, 3 of 4 developers will lead with sustainability as their primary development principle.

This is an ideal opportunity for a resource section on MDN for tools, strategies, standards, and impacts of developing more environmentally friendly code - optimizing images, minimizing unnecessary compute, caching, creating low-bandwidth page design, [insert your idea here].

[Cloud compute](https://www.ijert.org/survey-on-energy-consumption-in-cloud-computing), [AI](https://devblogs.microsoft.com/sustainable-software/the-carbon-footprint-of-ai/), [blockchain](https://digiconomist.net/bitcoin-energy-consumption), and other energy-hungry computational tasks are predicted to vastly increase in the coming years. Keeping these services running will require businesses and individuals to have more effective strategies and tools for managing energy consumption in order to meet climate objectives and mitigate rising energy costs, restrictions, and scarcities.

**Comment by Rumyra** [comment]
Thanks @heaversm - it seems there is scope for a guide here. We'll be spending the next few weeks on creating a roadmap for the team - once that's done I'll revisit this so it can be triaged 👍
